TalkAloud/TalkAloud/ViewControllers/AudioPlayerViewController.swift

```diff
@@ -24,12 +24,26 @@ class AudioViewController: UIViewController, AudioEngineStateChangeDelegate {
         recordAudioButton.isEnabled = true
     }
     
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        // Doing this on ViewWillAppear because in the future
+        // when I have the option to delete recording from the
+        // Audio Player View I will know that this check will always
+        // happen
+        if audioManager.isArrayEmpty() {
+            playAudioButton.isEnabled = false
+        } else {
+            playAudioButton.isEnabled = true
+        }
+        didUpdateAudioState(with: AudioEngine.sharedInstance.audioState)
```

Comment:
You are short circuiting stuff, adding code that is not meant to be added. 
This is not the solution.

Why would your class itself call that delegate method? It doesn't make sense.

Your problem is elsewhere.